blackjax/smc/inner_kernel_tuning.py

```diff
@@ -28,8 +30,11 @@ def build_kernel(
     mcmc_step_fn: Callable,
     mcmc_init_fn: Callable,
     resampling_fn: Callable,
-    mcmc_parameter_update_fn: Callable[[SMCState, SMCInfo], Dict[str, ArrayTree]],
+    mcmc_parameter_update_fn: Callable[
+        [PRNGKey, SMCState, SMCInfo], Dict[str, ArrayTree]
+    ],
     num_mcmc_steps: int = 10,
+    smc_returns_state_with_parameter_override=False,
```

Comment:
Is it possible to automatically detect this using the return from `mcmc_parameter_update_fn`

Reply:
it would be both the input and the return of smc_algorithm().step(), but we need to know them before executing a step (since the input is adapted too). 
I thought about having a decorator based registry of SMC implementations that tune MCMC parameters, so we would check belonging to that registry, but the problem is that any partial application or wrapper wouldn't work out of the box, the user would need to register it.

Reply:
In other words I don't know how to change while keeping the functional approach and without having the users annotating their functions somehow.

Reply:
thanks Carlos, I cannot think of a better alternative as well.